crates/net/downloaders/src/optimistic/download/bodies.rs

```diff
@@ -0,0 +1,44 @@
+use futures::Future;
+use reth_eth_wire::BlockBody;
+use reth_interfaces::p2p::error::PeerRequestResult;
+use std::{
+    pin::Pin,
+    sync::Arc,
+    task::{Context, Poll},
+};
+
+type BodiesFut = Pin<Box<dyn Future<Output = PeerRequestResult<Vec<BlockBody>>> + Send>>;
+
+/// TODO:
+pub(crate) struct BodiesDownload<B, C> {
+    client: Arc<B>,
+    consensus: Arc<C>,
+    fut: Option<BodiesFut>,
+}
+
+impl<H, C> std::fmt::Debug for BodiesDownload<H, C> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("BodiesDownload").finish()
+    }
+}
+
+impl<B, C> BodiesDownload<B, C> {
```

Comment:
I think technically these futures are up to the client that the downloader has a handle to, right? We don't need retry logic anymore really since we should retry forever, and we should not backoff because we can't know where the request is routed to